### PR TITLE
Martbern/fix/performance_by_threshold

### LINKED ISSUE #116: Table: Peformance by threshold
Seems like some of the metrics are wrong.
- [ ] What's up with the number of cases? 2x2-table seems wrong.
- [ ] Warning days are off?

### LINKED ISSUE #136: Regen dataset

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Fixes #116
Fixes #136

### CODE REVIEW

**Comment by MartinBernstorff** [comment]
Sorry about the mix of changes! Primary features:
- [ ] Parallelise pytest on github actions only (parallelisation slows down single tests a ton, but speeds up doing a "run all" by 2x)
- [ ] Refactor config files, specifically make one only for unit testing
- [ ] Fix warning days being incorrectly calculated, and update tests accordingly

File: src/psycopt2d/config/data/all_sql.yaml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 n_training_samples: null  # (int, null): Number of training samples to use, defaults to null in which cases it uses all samples.
+table_name: psycop_t2d_2022_07_14_14_10
 drop_patient_if_outcome_before_date: 2011-01-01
 min_lookahead_days: 0 # (int): Drop all prediction times where (max timestamp in the dataset) - (current timestamp) is less than min_lookahead_days
 lookahead_days: 1826.25 # (float): Number of days from prediction time to look ahead for the outcome.
 pred_col_name_prefix: "pred_" # (str): prefix of predictor columns
 pred_timestamp_col_name: timestamp # (str): Column name for prediction times
-outcome_timestamp_col_name: timestamp_diabetes_any  # (str): Column name for outcome timestamps
+outcome_timestamp_col_name: timestamp_first_t2d  # (str): Column name for outcome timestamps
```
Comment (HLasse) [inline src/psycopt2d/config/data/all_sql.yaml:8]: much better name :+1:

File: src/psycopt2d/config/data/all_sql.yaml [hunk 2]
```diff
@@ -1,9 +1,10 @@
 n_training_samples: null  # (int, null): Number of training samples to use, defaults to null in which cases it uses all samples.
+table_name: psycop_t2d_2022_07_14_14_10
```
Comment (HLasse) [inline src/psycopt2d/config/data/all_sql.yaml:2]: Big fan of this - both for versioning of tables but also for generalizability to other outcomes

File: tests/test_performance_by_threshold.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            "accuracy": {0: 0.8599, 1: 0.5, 2: 0.4539},
+            "true_positives": {0: 508, 1: 2510, 2: 2764},
+            "true_negatives": {0: 85485, 1: 47487, 2: 42627},
+            "false_positives": {0: 9492, 1: 47490, 2: 52350},
+            "false_negatives": {0: 4515, 1: 2513, 2: 2259},
+            "total_warning_days": {0: 609757.0, 1: 2619787.0, 2: 4612729.0},
+            "warning_days_per_false_positive": {0: 64.2, 1: 55.2, 2: 88.1},
+            "mean_warning_days": {0: 1252, 1: 1332, 3: 1451},
+        },
+    )
```
Comment (HLasse) [inline tests/test_performance_by_threshold.py:67]: You can also write the values as a list so you don't have to write the indices. E.g. 
```py
"PPV" : [0.508, 0.502, 0.502]
```

File: tests/test_performance_by_threshold.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            "FPR": {0: 0.0999, 1: 0.5, 2: 0.5512},
+            "FNR": {0: 0.8989, 1: 0.5003, 2: 0.4497},
+            "accuracy": {0: 0.8599, 1: 0.5, 2: 0.4539},
+            "true_positives": {0: 508, 1: 2510, 2: 2764},
+            "true_negatives": {0: 85485, 1: 47487, 2: 42627},
+            "false_positives": {0: 9492, 1: 47490, 2: 52350},
+            "false_negatives": {0: 4515, 1: 2513, 2: 2259},
+            "total_warning_days": {0: 609757.0, 1: 2619787.0, 2: 4612729.0},
+            "warning_days_per_false_positive": {0: 64.2, 1: 55.2, 2: 88.1},
+            "mean_warning_days": {0: 1252, 1: 1332, 3: 1451},
```
Comment (HLasse) [inline tests/test_performance_by_threshold.py:65]:
```suggestion
            "mean_warning_days": {0: 1252, 1: 1332, 2: 1451},
```

**Review by HLasse — APPROVED** [review]
Great work! Super nice to have the table name generalized and faster testing (+ the performance metric of course). There's an empty `train_model_old.py` file that should be deleted.
Feel free to merge!

**Pull request merged**